### PR TITLE
Prevent Hello with empty profile rendering in navbar when user already loggedin

### PR DESCRIPTION
**Pull request by kotesh-arya**
### Issue:
https://github.com/Real-Dev-Squad/website-status/issues/746
### Description:
`Login with Github` button showing have been handled in the Navbar while refreshing the Pages, when user already Loggedin
### Dev Tested:
- [x] Yes

### Images/video of the change:
**Before:**

https://github.com/Real-Dev-Squad/website-status/assets/69259490/24548937-15bb-466a-86ff-aaf1cc0395fc

**After:**

https://github.com/Real-Dev-Squad/website-status/assets/69259490/1a95ab8b-fc42-4bca-8226-039bb9d3f4cf

### Test Coverage Stats:

image

### CODE REVIEW

File: src/components/taskDetails/index.tsx [hunk 1]
```diff
@@ -238,13 +239,17 @@ const TaskDetails: FC<Props> = ({ taskID }) => {
...
+                                <button
+                                    onClick={() =>
+                                        router.push(
+                                            `/progress/${taskID}?dev=true`
+                                        )
+                                    }
+                                >
+                                    Update Progress
+                                </button>
+                            )}
```
Comment (vinit717) [inline src/components/taskDetails/index.tsx:252]: why this changes reflect in this pr

Comment (kotesh-arya) [reply src/components/taskDetails/index.tsx:252]: Actually, there's another PR which is yet to be merged, once it gets merged, these changes will no more exist 👍🏻 
Here's the PR: https://github.com/Real-Dev-Squad/website-status/pull/616

File: src/components/taskDetails/index.tsx [hunk 2]
```diff
@@ -56,6 +56,8 @@ type Props = {
 
 const TaskDetails: FC<Props> = ({ taskID }) => {
     const router = useRouter();
+    const { query } = router;
+    const isDevModeEnabled = query.dev === 'true' ? true : false;
```
[the remark this answers is not in the document]
Comment (kotesh-arya) [reply src/components/taskDetails/index.tsx:60]: > If the possible values of `query.dev` are only 'true' and 'false' then you can choose to use the below code.
> 
> ```
> JSON.parse(query.dev);
> ```
> 
> This is just another way of converting 'true' and 'false' strings to respective booleans. It is up to you if you want to use this, the conditional operator code is understandable. If you use JSON.parse add a brief comment on what it does.

Thanks for the great suggestion @apurvkhare  🙌🏻 
But in this case I am targeting only for `query.dev` value being **true** when it has string 'true'  and in all the rest cases of `query.dev` will be false.

File: src/components/navBar/index.tsx [hunk 3]
```diff
@@ -78,40 +78,44 @@ const NavBar = () => {
                 </ul>
             </div>
             <div className={styles.userProfile}>
-                {!isLoggedIn ? (
-                    <Link href={LOGIN_URL}>
-                        <button className={styles.signInLink}>
-                            Sign In With Github
+                {userData !== undefined ? (
+                    userData ? (
```
Comment (TanishqSingla) [inline src/components/navBar/index.tsx:82]: if userData happens to be `{}` it will return true, make sure you're handling that case

File: src/components/navBar/index.tsx [hunk 4]
```diff
@@ -78,40 +78,44 @@ const NavBar = () => {
...
+                                    src={GITHUB_LOGO}
+                                    height="20"
+                                    width="20"
+                                    alt="GitHub Icon"
+                                />
+                            </button>
+                        </Link>
+                    )
+                ) : (
+                    ''
```
Comment (TanishqSingla) [inline src/components/navBar/index.tsx:118]: we can just rely on `&&` operator for conditional rendering, that way we won't have to print `''`

**Review by TanishqSingla — CHANGES_REQUESTED** [review]

File: src/components/taskDetails/index.tsx (hunk 1 above)
Comment (kotesh-arya) [reply src/components/taskDetails/index.tsx]: **P.S: ** Now those changes will no more reflect in this PR.

File: src/components/navBar/index.tsx (hunk 4 above)
Comment (kotesh-arya) [reply src/components/navBar/index.tsx]: Updated with Short Circuit Operator - **&&** 👍🏻

File: __tests__/Unit/Components/Navbar/Navbar.test.tsx [hunk 5]
```diff
@@ -91,4 +91,12 @@ describe('Navbar', () => {
         const logo = getByTestId('logo');
         expect(logo).toBeInTheDocument();
     });
+    test('Sign In With Github button should not render when user is already loggedin', async () => {
+        renderWithProviders(<NavBar />);
+        await waitFor(() => {
+            const loginTextElement = screen.queryByText('Sign In With Github');
+            expect(loginTextElement).not.toBeInTheDocument();
+            screen.debug();
```
[the remark this answers is not in the document]
Comment (kotesh-arya) [reply __tests__/Unit/Components/Navbar/Navbar.test.tsx]: Removed, Thanks for pointing 🤝

File: src/components/navBar/index.tsx (hunk 3 above)
Comment (kotesh-arya) [reply src/components/navBar/index.tsx]: The userData will either be undefined or an Object with Data properties right ?
Can you please elaborate on this empty Object { } case, Thank you

**Review by bksh05 — APPROVED** [review]
LGTM

File: src/components/navBar/index.tsx [hunk 6]
```diff
@@ -78,41 +78,42 @@ const NavBar = () => {
                 </ul>
             </div>
             <div className={styles.userProfile}>
-                {!isLoggedIn ? (
-                    <Link href={LOGIN_URL}>
-                        <button className={styles.signInLink}>
-                            Sign In With Github
+                {userData !== undefined &&
```
Comment (TanishqSingla) [inline src/components/navBar/index.tsx:81]: instead of checking equality with undefined, we can just use `userData` in this line, this will return a falsy value. If you want to be more specific about the boolean then you can use it with `!!userData` which will convert the Falsy value to `false`

**Review by TanishqSingla — CHANGES_REQUESTED** [review]

File: src/components/navBar/index.tsx (hunk 6 above)
Comment (kotesh-arya) [reply src/components/navBar/index.tsx]: Updated as per the suggestion 👍🏻

**Review by TanishqSingla — APPROVED** [review]

File: src/components/navBar/index.tsx [hunk 7]
```diff
@@ -78,41 +78,42 @@ const NavBar = () => {
...
-                            }
-                            alt="Profile Pic"
-                            width="32"
-                            height="32"
-                        />
-                        {toggleDropdown && <Dropdown />}
-                    </div>
-                )}
+                    ) : (
+                        <Link href={LOGIN_URL}>
```
Comment (shubhamsigdar1) [inline src/components/navBar/index.tsx:104]: When it will run??

**Comment by kotesh-arya** [comment]
Closing this request as this issue had been addressed in another PR https://github.com/Real-Dev-Squad/website-status/pull/728
